### ISSUE TITLE
Go generating non-compiling bits

### ISSUE THREAD

**Issue by jadekler** [googleapis/gapic-generator#2676]
See: https://code-review.googlesource.com/c/gocloud/+/39510/3/firestore/apiv1beta1/firestore_client.go

```
// Write streams batches of document updates and deletes, in order.
func (c *Client) Write(ctx context.Context, opts ...gax.CallOption) (firestorepb.Firestore_WriteClient, error) {
	md := metadata.Pairs("x-goog-request-params", fmt.Sprintf("%s=%v", "database", req.GetDatabase()))
	ctx = insertMetadata(ctx, c.xGoogMetadata, md)
```

It tries to use a `req`, which doesn't exist.

**Comment by andreamlin** [comment]
I had submitted a bug fix for this, but it might not have solved the problem.

Was this created using the latest gapic-generator / artman (0.16.22)?

**Comment by jadekler** [comment]
>Was this created using the latest gapic-generator / artman (0.16.22)?
Yep!

**Comment by andreamlin** [comment]
I'll definitely look at it again. For the time being, I think 0.16.20 doesn't have this bug.

**Comment by andreamlin** [comment]
was this the only method and the only affected API client?

**Comment by andreamlin** [comment]
update to worklog: this works fine if i use artman with gapic-generator at master, but doesn't work if i use the artman docker image by itself. i'm guessing that the bug fix didn't make it out to the new artman image. will try re-releasing the artman image

**Comment by andreamlin** [comment]
this should be fixed in artman 0.16.23.

After getting the new image, i ran `artman --config google/firestore/artman_firestore_v1.yaml generate go_gapic`

and got, among other files, a firestore_client.go that includes the following code snippet
```
// Write streams batches of document updates and deletes, in order.
func (c *Client) Write(ctx context.Context, opts ...gax.CallOption) (firestorepb.Firestore_WriteClient, error) {
        ctx = insertMetadata(ctx, c.xGoogMetadata)
        opts = append(c.CallOptions.Write[0:len(c.CallOptions.Write):len(c.CallOptions.Write)], opts...)
        var resp firestorepb.Firestore_WriteClient
        err := gax.Invoke(ctx, func(ctx context.Context, settings gax.CallSettings) error {
                var err error
                resp, err = c.client.Write(ctx, settings.GRPC...)
                return err
        }, opts...)
```

@jadekler Does this solve the problem?

**Comment by jadekler** [comment]
Looks good in 0.16.23! Thanks!
